test/helpers/PermalinksTest.ts

```diff
@@ -125,26 +129,37 @@ describe('Permalinks', () => {
             const parsed = Permalinks.parseUrl(`https://matrix.to/#/${roomId}/${eventId}`);
 
             expect(parsed).toMatchObject(<any>expected);
+            expect(Permalinks.parseUrl(Permalinks.forEvent(roomId, eventId))).toMatchObject(expected);
         });
 
         it('should parse room alias permalink URLs with via servers', () => {
             const roomId = "#test:example.org";
             const eventId = "$ev:example.org";
             const via = ["one.example.org", "two.example.org"];
             const expected: PermalinkParts = { userId: undefined, roomIdOrAlias: roomId, viaServers: via, eventId };
-            const parsed = Permalinks.parseUrl(`https://matrix.to/#/${roomId}/${eventId}?via=${via.join("via=")}`);
-
+            const parsed = Permalinks.parseUrl(`https://matrix.to/#/${roomId}/${eventId}?via=${via.join("&via=")}`);
             expect(parsed).toMatchObject(<any>expected);
+            expect(Permalinks.parseUrl(Permalinks.forEvent(roomId, eventId, via))).toMatchObject(expected);
         });
 
         it('should parse room ID permalink URLs with via servers', () => {
             const roomId = "!test:example.org";
             const eventId = "$ev:example.org";
             const via = ["one.example.org", "two.example.org"];
             const expected: PermalinkParts = { userId: undefined, roomIdOrAlias: roomId, viaServers: via, eventId };
-            const parsed = Permalinks.parseUrl(`https://matrix.to/#/${roomId}/${eventId}?via=${via.join("via=")}`);
+            const parsed = Permalinks.parseUrl(`https://matrix.to/#/${roomId}/${eventId}?via=${via.join("&via=")}`);
 
             expect(parsed).toMatchObject(<any>expected);
+            expect(Permalinks.parseUrl(Permalinks.forEvent(roomId, eventId, via))).toMatchObject(expected);
+        });
+
+        it('should parse room ID permalink URLs with via servers', () => {
```

Comment:
```suggestion
        it('should parse room ID URLs with via servers', () => {
```

(permalinks are technically ones with event IDs)